src/adapter/src/coord/sequencer/alter_set_cluster.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use mz_expr::CollectionPlan;
+use mz_sql::catalog::ObjectType;
+use mz_sql::plan::AlterSetClusterPlan;
+use std::iter::once;
+
+use crate::catalog::CatalogItem;
+use crate::coord::Coordinator;
+use crate::session::Session;
+use crate::{catalog, AdapterError, ExecuteResponse, TimestampProvider};
+
+impl Coordinator {
+    /// Convert a [`AlterSetClusterPlan`] to a sequence of catalog operators and adjust state.
+    pub(super) async fn sequence_alter_set_cluster(
+        &mut self,
+        session: &Session,
+        AlterSetClusterPlan { id, set_cluster }: AlterSetClusterPlan,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        let entry = self.catalog().get_entry(&id);
+        match entry.item() {
+            CatalogItem::MaterializedView(catalog::MaterializedView {
+                optimized_expr,
+                cluster_id,
+                ..
+            }) => {
+                if !self
+                    .catalog()
+                    .state()
+                    .is_system_schema_specifier(&entry.name().qualifiers.schema_spec)
```

Comment:
What's the purpose of this check? Do we store system MVs on storage clusters? (I wasn't even aware that system MVs exist.)

Reply:
Ah, this is copied from `sequence_create_materialized_view`. Perhaps this was copied without much thought from `sequence_create_index`.

Reply:
Removed.